products/BellHybrid/apps/application-bell-whats-new/models/WhatsNewFeaturesModel.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2017-2024, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "WhatsNewFeaturesModel.hpp"
+#include <ApplicationCommon.hpp>
+#include <db/ServiceDB.hpp>
+#include <db/WhatsNewMessages.hpp>
+#include <service-db/Settings.hpp>
+#include <service-db/agents/settings/SystemSettings.hpp>
+#include <service-desktop/Constants.hpp>
+#include <module-utils/utility/Version.hpp>
+#include <product/version.hpp>
+#include <Utils.hpp>
+
+namespace
+{
+    using namespace service::db::whatsnew;
+
+    auto getVersionNumber(const std::string &version) -> std::optional<VersionNumber>
+    {
+        constexpr auto versionSize{3U};
+
+        std::vector<std::string> strVector{utils::split(version, '.')};
+        if (strVector.size() != versionSize) {
+            return {};
+        }
+
+        std::vector<std::uint16_t> uintVector{};
+        uintVector.reserve(versionSize);
+
+        for (const auto &str : strVector) {
+            if (!utils::is_number(str)) {
+                return {};
+            }
+            uintVector.push_back(utils::getNumericValue<std::uint16_t>(str));
+        }
+        return VersionNumber{uintVector[0], uintVector[1], uintVector[2]};
+    }
+
+    auto sendDBRequest(sys::Service *serv, std::shared_ptr<sys::Message> &&msg) -> std::optional<messages::Response>
+    {
+        const auto ret = serv->bus.sendUnicastSync(std::move(msg), service::name::db, sys::BusProxy::defaultTimeout);
+        if (ret.first == sys::ReturnCodes::Success) {
+            if (auto resp = std::dynamic_pointer_cast<messages::Response>(ret.second)) {
+                return *resp;
+            }
+        }
+        return {};
+    }
+} // namespace
+
+namespace app::whatsnew::models
+{
+    WhatsNewFeaturesModel::WhatsNewFeaturesModel(app::ApplicationCommon *app, settings::Settings *settings)
+        : settings{settings}
+    {
+        auto previousVersion =
+            settings->getValue(settings::SystemProperties::osCurrentVersion, settings::SettingsScope::Global);
+        if (previousVersion == utils::defaultPreviousVersion) {
+            previousVersion = utils::getPreviousVersionFromFile(purefs::dir::getTemporaryPath() /
```

Comment:
If it's either a valid string or `utils::defaultPreviousVersion` why not return `utils::defaultPreviousVersion` from `getPreviousVersionFromFile` as a fallback value instead of an empty optional? The name suggests it should be done this way 🙂 Not a blocker, though!

Reply:
Good point! IMO the implementation returning `std::optional` is more flexible, as it allows to distinguish between failure of retrieving version data from the file and retrieving the value that happens to be the same that the fallback one. In this case it doesn't matter at all, but if this function is used in some other context in the future this additional info might be useful...and if it's not, then at least such implementation doesn't introduce much overhead :wink: